src/index.js

```diff
@@ -33,3 +64,26 @@ if ( document.readyState !== 'loading' ) {
   document.addEventListener( 'DOMContentLoaded', onReady );
 }
 
+/* 
+REFLECTION
+
+Challenges: 
+	* Knowing what is already in the DOM and what is being added. in renderTheUI(), my instinct 
```

Comment:
These are all good thoughts. One feedback I would provide is to avoid premature optimization.

Where do you think a framework fits in here? How do they solve some of these challenges?